### ISSUE TITLE
.aac and .adts files not showed

### ISSUE THREAD

**Issue by RafaelLinux** [keithgg/puddletag#406]
I have some files with extension .aac or .adts
If I don't change their file extension, they are player correctly in Audacious.
However, Puddletag do not see that files even if I rename they to ".m4a" extension (and then Audacious can't play them).
What's the problem? Why they are not showed in Puddletag in any way?

**Comment by thwaller** [comment]
#374 This looks similar to a report I filed a while back regarding AAC in MP4. I also use Audacious, so I thought to reply as our environments are at least similar.

I can resolve this specific error (as I reported it) by using the source from puddletag-master and using that. I am using Ubuntu MATE 18.04, and my issue is gone with AAC in MP4 with using that.

**Comment by jlu5** [comment]
AFAIK .m4a and .aac files are two distinct formats - files using the .aac extension include only the raw audio data, while .m4a refers to the [MP4 container](https://en.wikipedia.org/wiki/MPEG-4_Part_14#Filename_extensions) commonly placed around AAC streams (the MP4 container is what supports metadata).

You should be able to [losslessly convert](https://ffmpeg.org/ffmpeg.html#Stream-copy) .aac and .adts files to .m4a using a tool like ffmpeg: `ffmpeg -i input.aac -c copy output.m4a`. Then you should be able to write metadata to it

**Comment by thwaller** [comment]
Yes, the m4a is the mp4 container (as well as mp4a, m4p, m4v, m4b, etc) and aac is the audio encoding most commonly contained within. The original poster is asking about Puddletag not seeing the files though, not what format is what (although just changing the extension from aac to m4a is not a valid action). From my experience with this issue, the issue is resolved using from puddletag-master. There was a fix(s) done that do not appear in the repos for Ubuntu at this time, but you can get the fix via source here.

My suggestion is an attempt to help the poster resolve his/her issues, which I believe it will, as it relates to both audio files with the m4a container/extension as well as aac.

**Comment by RafaelLinux** [comment]
Thank you to both. The problem is that my carplayer (Kenwood) only recognizes .aac audio files, so to preserve that format is a must. Anyway, I had curiosity about metadata in .aac files and found this:

> A KDE/Qt application for Linux/Unix called easytag will edit metadata about AAC-encoded files as long as they end with the .m4a extension. Typically, renaming files to .m4a and back to the original extension after editing is sufficient. This method can be used to configure typical ID3v2 metadata including cover art. This cover art is used by Apple's iTunes and their media players. Note that easytag does not have the ability to add chapter points into Audiobooks (.m4b).

I'll just try it tonight with easytag, but obviously I prefer puddletag, cause I knew easy tag too and for me, puddletag it's superior.

**Comment by RafaelLinux** [comment]
Thank you to both. The problem is that my carplayer (Kenwood) only recognizes .aac audio files, so to preserve that format is a must. Anyway, I had curiosity about metadata in .aac files and found this:

> A KDE/Qt application for Linux/Unix called easytag will edit metadata about AAC-encoded files as long as they end with the .m4a extension. Typically, renaming files to .m4a and back to the original extension after editing is sufficient. This method can be used to configure typical ID3v2 metadata including cover art. This cover art is used by Apple's iTunes and their media players. Note that easytag does not have the ability to add chapter points into Audiobooks (.m4b).

I'll just try it tonight with easytag, but obviously I prefer puddletag, cause I knew easy tag too and for me, puddletag it's superior.

Note: MP3TAG pro (Windows app) let change metadata of .aac files, so yes, it's possible!!! Will puddletag add this feature?

**Comment by thwaller** [comment]
> Note: MP3TAG pro (Windows app) let change metadata of .aac files, so yes, it's possible!!! Will puddletag add this feature?

Did you try running from source vs PPA? The AAC ext does support ID3 tags.

**Comment by RafaelLinux** [comment]
I'm using OpenSUSE, so my only chance to try what you said it to compile from source, and I'm afraid I'm not very good on compiling apps ....  :(

**Comment by thwaller** [comment]
I do not know OpenSUSE well at all, but if it is the same, there is no need for compile. You just download the zip and "You can run puddletag from that directory by typing ./puddletag in your console.".
